### PR TITLE
Remove all references to cmake for the builds in this repo

### PR DESCRIPTION
**Pull request by TravisEz13**
# PR Summary

Remove all references to cmake for the builds in this repo

## PR Context

Consider backporting this so we don't install build tools we don't need.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

File: test/perf/benchmarks/assets/compiler.test.ps1 [hunk 1]
```diff
@@ -1149,7 +1149,7 @@ function Start-PSBootstrap {
             $Deps = @()
             if ($environment.IsLinux -and $environment.IsUbuntu) {
                 # Build tools
-                $Deps += "curl", "g++", "cmake", "make"
+                $Deps += "curl", "g++", "make"
```
Comment (adityapatwardhan) [inline test/perf/benchmarks/assets/compiler.test.ps1:1152]: do we need g++ and make?

File: test/perf/benchmarks/assets/compiler.test.ps1 [hunk 2]
```diff
@@ -1179,7 +1179,7 @@ function Start-PSBootstrap {
                 }
             } elseif ($environment.IsLinux -and $environment.IsRedHatFamily) {
                 # Build tools
-                $Deps += "which", "curl", "gcc-c++", "cmake", "make"
+                $Deps += "which", "curl", "gcc-c++", "make"
```
Comment (adityapatwardhan) [inline test/perf/benchmarks/assets/compiler.test.ps1:1182]: same as above

File: test/perf/benchmarks/assets/compiler.test.ps1 [hunk 3]
```diff
@@ -1203,7 +1203,7 @@ function Start-PSBootstrap {
                 }
             } elseif ($environment.IsLinux -and $environment.IsSUSEFamily) {
                 # Build tools
-                $Deps += "gcc", "cmake", "make"
+                $Deps += "gcc", "make"
```
Comment (adityapatwardhan) [inline test/perf/benchmarks/assets/compiler.test.ps1:1206]: same

**Review by adityapatwardhan — CHANGES_REQUESTED** [review]

File: test/perf/benchmarks/assets/compiler.test.ps1 (hunk 1 above)
Comment (daxian-dbw) [reply test/perf/benchmarks/assets/compiler.test.ps1:1152]: This file is a test file, so the content in it doesn't matter.

File: test/perf/benchmarks/assets/compiler.test.ps1 (hunk 2 above)
Comment (daxian-dbw) [reply test/perf/benchmarks/assets/compiler.test.ps1:1182]: same here. this file is for performance testing.

File: test/perf/benchmarks/assets/compiler.test.ps1 (hunk 3 above)
Comment (daxian-dbw) [reply test/perf/benchmarks/assets/compiler.test.ps1:1206]: ditto. changes to this file can be reverted.

**Review by adityapatwardhan — APPROVED** [review]

**Comment by TravisEz13** [comment]
cherry-pick to 7.0 resulted in no changes

**Comment by TravisEz13** [comment]
96bdcfca8cb28e58b3c45eedc94864bed6360ba3 is 7.0 commit to backport

**Pull request merged**